types/src/on_chain_config/consensus_config.rs

```diff
@@ -241,6 +244,48 @@ impl OnChainConsensusConfig {
     pub fn is_vtxn_enabled(&self) -> bool {
         self.effective_validator_txn_config().enabled()
     }
+
+    pub fn disable_validator_txns(&mut self) {
+        match self {
+            OnChainConsensusConfig::V1(_) | OnChainConsensusConfig::V2(_) => {
+                // vtxn not supported. No-op.
+            },
+            OnChainConsensusConfig::V3 { vtxn, .. } => {
+                *vtxn = ValidatorTxnConfig::V0;
+            },
+        }
+    }
+
+    pub fn enable_validator_txns(&mut self) {
```

Comment:
why this is a mutation? how do we use this function?

Reply:
later in smoke tests, example [here](https://github.com/aptos-labs/aptos-core/blob/d78f6902b2e2076309c2d391a9346ed151c6a65a/testsuite/smoke-test/src/randomness/enable_feature_1.rs#L28)